program/src/state/rule_set.rs

```diff
@@ -13,7 +10,7 @@ pub struct RuleSet {
     /// Owner (creator) of the RuleSet.
     owner: Pubkey,
     /// A map to determine the `Rule` that belongs to a given `Operation`.
-    pub operations: HashMap<Operation, Rule>,
+    pub operations: HashMap<u16, Rule>,
```

Comment:
Also, while you are making this change, could you add a more expressive error to processor.rs for when the key cannot be found in this code?

```
// Get the Rule from the RuleSet based on the caller-specified Operation.
let rule = rule_set
    .get(args.operation)
    .ok_or(RuleSetError::DataTypeMismatch)?;
```
I think it should be something like `OperationNotFound`.